tests/vertexonly/test_poisson_inverse_conductivity.py

```diff
@@ -29,12 +29,6 @@ def test_poisson_inverse_conductivity():
     # Have to import inside test to make sure cleanup fixtures work as intended
     from firedrake_adjoint import Control, ReducedFunctional, minimize
 
-    # Manually set up annotation since test suite may have stopped it
```

Comment:
Quick query before I approve, why was this changed?

Reply:
With [this line](https://github.com/firedrakeproject/firedrake/blob/master/tests/vertexonly/test_poisson_inverse_conductivity.py#L18) above it in the file we were effectively switching on the adjoint twice. Since we use a [counter](https://github.com/dolfin-adjoint/pyadjoint/blob/master/pyadjoint/tape.py#L31) to track whether or not the adjoint is enabled this meant that switching the adjoint off with `with stop_annotating():` only incremented the counter once and we still thought that we were supposed to be taping.

This was causing tests to fail because I now `raise NotImplementedError` if someone is inside either `FormAssembler.assemble` or `Assigner.assign` and taping is switched on.

Reply:
But there is no `with stop_annotating():` here? Or am I being daft?

Reply:
We call `with stop_annotating():` inside the adjoint (e.g. [here](https://github.com/firedrakeproject/firedrake/blob/master/firedrake/adjoint/assembly.py#L18)). I think the idea is to make sure that we aren't adding blocks to the tape that are 'inside' other blocks. An example would be that we want to tape a `solve` block but not the `assemble` calls that happen inside the `solve`.

Reply:
Okay I think I follow. Either way it's nice to see that this "just in case" code can be got rid of! I think previously it wasn't clear in either case where we needed it so it's good that now you got an error thrown